sdks/android/Mentat/library/src/main/java/org/mozilla/mentat/Query.java

```diff
@@ -281,22 +284,23 @@ public void run(final CollResultHandler handler) {
      * @param handler   the handler to call with the results of this query
      */
     public void run(final TupleResultHandler handler) {
-        RustResult result = JNA.INSTANCE.query_builder_execute_tuple(this.consumePointer());
+        RustError.ByReference error = new RustError.ByReference();
```

Comment:
When we finally get around to throwing exceptions on error in Java rather than just logging and moving on, is this going to affect this pattern so we have to have something similar to Swift?

Reply:
I think the error.logOrWhatever() will just be `error.throwIfError()`. The lack of lightweight first class functions in java means that IDK if it's worth trying to abstract it away, especially given that the abstraction would increase the amount of code callers have to type compared to doing it directly.